### PR TITLE
DOCS-2801: Add flutter sample code to control motor quickstart

### PR DESCRIPTION
**Pull request by sguequierre**
* instructs them to go through making flutter app process, then adds a motor_screen and edits the robot_screen to include the motor screen
* adds picture of motor screen from simulator

### CODE REVIEW

File: docs/get-started/control-motor.md [hunk 1]
```diff
@@ -244,6 +244,197 @@ func main() {
...
+                    onTap: () => _navigate(resourceName),
+                    trailing: _isNavigable(resourceName) ? Icon(Icons.chevron_right) : SizedBox.shrink(),
+                  );
+                }));
+  }
+}
+```
+
+This imports the <file>motor_screen.dart</file> file into the program and adds logic to check if a {{< glossary_tooltip term_id="resource" text="resource" >}} is "navigable", or, has a screen made for it.
+Base is the only resource that is navigable.
```
Comment (sguequierre) [inline docs/get-started/control-motor.md:429]:
```suggestion
Motor is currently the only resource that is navigable.
```

File: docs/get-started/control-motor.md [hunk 2]
```diff
@@ -244,6 +244,197 @@ func main() {
...
+                    onTap: () => _navigate(resourceName),
+                    trailing: _isNavigable(resourceName) ? Icon(Icons.chevron_right) : SizedBox.shrink(),
+                  );
+                }));
+  }
+}
+```
+
+This imports the <file>motor_screen.dart</file> file into the program and adds logic to check if a {{< glossary_tooltip term_id="resource" text="resource" >}} is "navigable", or, has a screen made for it.
+Motor is currently the only resource that is navigable.
```
Comment (JessamyT) [inline docs/get-started/control-motor.md:429]: "currently..." sounds almost as if Viam currently only supports Flutter motor pages, or something. How about rewording, perhaps:
```suggestion
Since you added a screen for motor, motor is a navigable resource.
```

File: docs/get-started/control-motor.md [hunk 3]
```diff
@@ -244,6 +244,197 @@ func main() {
 }
 ```
 
+{{% /tab %}}
+{{% tab name="Flutter" %}}
+
+Flutter code must be launched from inside a running Flutter application.
+To get started programming your rover with Flutter, follow the instructions to [Build a Flutter App that Integrates with Viam](/tutorials/control/flutter-app/).
```
Comment (JessamyT) [inline docs/get-started/control-motor.md:251]: I would not make them go through this entire tutorial. They don't need a locations screen etc. for a quickstart. https://docs.viam.com/get-started/drive-rover/ can also be simplified. When you tested this code, did you go through the whole flutter app tutorial or were you able to do a simpler subset of those instructions? I believe I've made test apps where I made the "home" screen after login just be the machine screen with its resources, and then make one or more of them navigable, so they auth to one machine but don't need to build the org and location screens. Of course the code needs tweaking it's not terribly complex, and those files could be pasted into this doc.

As-is, the Flutter tab is drastically more complex than the other languages, and it shouldn't be too hard to change the code and say "follow xyz steps of the flutter tutorial" instead of follow the whole thing.

If this does stay as-is, are there really no changes that need to be made to the robot screen code to enable this navigation? I don't remember off hand, just double checking that that's been tested.

Comment (sguequierre) [reply docs/get-started/control-motor.md:251]: You do need to edit the robot screen code, see the second part of this code. 

Re. not going through whole flutter app, I can try but it will take me a while. I'm not as familiar with flutter. I had already built the flutter app for the base move in square tutorial so I used that but tweaked for just the motor code/screens. I think it would be a lot of work to instruct them to build a flutter app entirely in this space but if you think that's the go ahead I can give it a shot. It might help me if you were to share one of those test apps?

Comment (JessamyT) [reply docs/get-started/control-motor.md:251]: Ooh yes I see I was looking at the full code section and not realizing, sorry. Okay cool.

So, I don't have a test app ready to share, because I edited mine adding lots of pages for lots of resources when I was testing the methods for the flutter SDK code example snippets. So it's not clean, and might be out of date. I was just figuring things out like you would be; hadn't used Flutter before other than editing/putting together the flutter tutorial based on Clint's doc. But, if you think this is too much work for now would you mind making a ticket to revisit this for this and the square one? I think it's a good idea not just for simplicity of these quickstarts but also to show a different construction of an app. TY!

Comment (JessamyT) [reply docs/get-started/control-motor.md:251]:
```suggestion
To get started programming your rover with Flutter, follow the instructions to [Build a Flutter App that Integrates with Viam](/tutorials/control/flutter-app/).
Then return to this page to add motor control to your app.
```
Or something to make it clear that they should do that and then continue here.

File: docs/get-started/control-motor.md [hunk 4]
```diff
@@ -244,6 +244,197 @@ func main() {
...
+                }));
+  }
+}
+```
+
+This imports the <file>motor_screen.dart</file> file into the program and adds logic to check if a {{< glossary_tooltip term_id="resource" text="resource" >}} is "navigable", or, has a screen made for it.
+Since you added a screen for motor, motor is a navigable resource.
+
+To navigate to the motor screen, save your code and launch your simulator.
+Navigate to the robot screen of your (live) machine with a motor resource configured, and see the resource displayed like the following:
```
Comment (JessamyT) [inline docs/get-started/control-motor.md:432]: "like the following" sounds awk to me
```suggestion
Navigate to the robot screen of your (live) machine with a motor resource configured, and see the resource control interface displayed:
```

**Review by JessamyT — APPROVED** [review]
LGTM % two small comments

File: docs/get-started/control-motor.md (hunk 3 above)
Comment (npentrel) [reply docs/get-started/control-motor.md:251]: I'm not sure I fully understand - so does this work without it being a rover, i.e. without having a base?

Comment (sguequierre) [reply docs/get-started/control-motor.md:251]: it should, the flutter app tutorial doesn't add any rover specific code, just a home screen and robot screen. This was in error, thank you for catching (my eyes confused rover->robot)

Comment (JessamyT) [reply docs/get-started/control-motor.md]: Sorry, the rover -> machine is not the important part of my comment. The important part is that I think we should say "Then return to this page to add motor control to your app." And yeah, you can 1000% have a motor without it being a rover. This isn't the drive in square quickstart, it's just a motor.
